qiita_pet/handlers/study_handlers/sample_template.py

```diff
@@ -57,3 +66,80 @@ def post(self):
         else:
             raise HTTPError(400, 'Unknown sample template action: %s' % action)
         self.write(result)
+
+
+def _build_sample_summary(study_id, user_id):
+    """Builds the initial table of samples associated with prep templates
+
+    Parameters
+    ----------
+    study_id : int
+        Study to get samples from
+    user_id : str
+        User requesting the information
+
+    Returns
+    -------
+    str
+        HTML of the summary table
+    """
+    # Load all samples available into dictionary and set
+    samps_table = {s: [] for s in
+                   sample_template_samples_get_req(
+        study_id, user_id)['samples']}
+    all_samps = set(samps_table.keys())
+    col_names = ['']
+    # Add one column per prep template highlighting what samples exist
+    preps = study_prep_get_req(study_id, user_id)['info']
+    for dt in preps:
+        for prep in preps[dt]:
+            prep_samples = prep_template_samples_get_req(
+                prep['id'], user_id)['samples']
+            # Empty cell for samples not in the prep template
+            for s in all_samps.difference(prep_samples):
+                samps_table[s].append('<td></td>')
+            # X in cell for samples in the prep template
+            for s in all_samps.intersection(prep_samples):
+                samps_table[s].append('<td>X</td>')
+            col_names.append('%s - %d' % (prep['name'], prep['id']))
+
+    # build the table header
+    table = ('<table id="samples-table" class="table table-striped"><thead>'
```

Comment:
I just want to bring up the fact that this is not going to work very well and that rendering a table of this size all at once is excessive, there are other ways of rendering the data, for example using clusterize http://www.plugingeek.com/repos/NeXTs/Clusterize.js